src/app/course/add-page/add-course-container/add-course-container.component.ts

```diff
@@ -9,24 +11,59 @@ import { CourseService } from '../../course.service';
   styleUrls: ['./add-course-container.component.scss']
 })
 export class AddCourseContainerComponent implements OnInit {
-  course: CourseInstance;
-  constructor(public coursesService: CourseService, private router: Router,  private route: ActivatedRoute) { } 
+  course: Course;
+  mode: string;
+  title: string;
+  newIndex;
+  form: FormGroup;
+
+  constructor(public coursesService: CourseService, private router: Router,  private route: ActivatedRoute) { 
+    this.route.params.pipe(
+      tap(data => {
+        console.log("DATA", data)
+        return data;
+      })
+    ).subscribe(data => {
+      console.log("Subsc data", data);
+      if (data.id) {
+        this.mode = 'Edit';
+        this.title = 'Edit course';
+        return;
+      }
+      this.mode = 'Add';
+      this.title = "Add new course"
+    })
+  } 
 
   ngOnInit(): void {
-    this.route.paramMap.subscribe((data) => {
-      const id = +data.get('id');
-      this.getCourseDetails(id);
-    });
+    this.createForm();
     this.createNewCourse();
   }
 
+
+
+  createForm(): void {
+    this.form = new FormGroup({
```

Comment:
`FormBuilder` simplifies form creation. It's going to be covered in the next modules of this course.